### PR TITLE
#3983 - CAS Addressline1 Update - Restrict to Alpha Numeric

### PR DESCRIPTION
**Pull request by andrepestana-aot**
- Added `replace(/[^A-Z0-9\-\s]/g, "")` to `formatAddress()` to remove everything that is not A-Z (capital letters), hyphens or whitespace;
- Added invalid characters to unit test.

### CODE REVIEW

File: sources/packages/backend/libs/integrations/src/cas/cas-formatters.ts [hunk 1]
```diff
@@ -26,9 +26,10 @@ export function formatUserName(firstName: string, lastName: string): string {
  * @returns formatted address.
  */
 export function formatAddress(address: string): string {
-  return convertToASCIIString(
-    address.substring(0, CAS_ADDRESS_MAX_LENGTH).trim(),
-  ).toUpperCase();
+  return convertToASCIIString(address)
+    .toUpperCase()
+    .replace(/[^A-Z0-9\-\s]/g, "")
```
Comment (guru-aot) [inline sources/packages/backend/libs/integrations/src/cas/cas-formatters.ts:31]: 👍

File: sources/packages/backend/libs/integrations/src/cas/cas-formatters.ts [hunk 2]
```diff
@@ -26,9 +26,10 @@ export function formatUserName(firstName: string, lastName: string): string {
...
+    .substring(0, CAS_ADDRESS_MAX_LENGTH);
```
Comment (guru-aot) [inline sources/packages/backend/libs/integrations/src/cas/cas-formatters.ts:32]: Do we need to add trim at the end?

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @andrepestana-aot . Minor comment

**Review by andrewsignori-aot — APPROVED** [review]
Great work, looks good 👍

**Pull request merged**